flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/collect/CollectSinkOperatorCoordinatorTest.java

```diff
@@ -105,54 +102,147 @@ void testCloseAfterRequestIsReceived() throws Exception {
                             createRequestForServerGeneratedResponse());
-            socketServer.handleRequest(expectedData1);
-            assertResponseWithDefaultMetadataFromServer(responseFuture1, expectedData1);
+            assertThat(responseFuture).isNotDone();
+
+            socketServer.handleRequest(expectedData);
+
+            assertResponseWithDefaultMetadataFromServer(responseFuture, expectedData);
+        }
+    }
+
+    @Test
+    void testServerSideClosingTheServerSocket() throws Exception {
+        try (CollectSinkOperatorCoordinator coordinator = new CollectSinkOperatorCoordinator()) {
+            coordinator.start();
+
+            final CompletableFuture<CoordinationResponse> responseFuture;
+            try (final TestingSocketServer socketServer =
+                    TestingSocketServer.createSocketServerAndInitializeCoordinator(coordinator)) {
+                final String version = "version";
+                responseFuture =
+                        coordinator.handleCoordinationRequest(
+                                createRequestForClientGeneratedResponse(version));
+                assertThat(responseFuture).isNotDone();
+
+                FlinkAssertions.assertThatFuture(socketServer.handleRequestWithoutResponse())
+                        .eventuallySucceeds();
+            }
+            assertEmptyResponseGeneratedFromServer(responseFuture);
+        }
+    }
+
+    @Test
+    void testServerSideClosingTheAcceptingSocket() throws Exception {
+        try (CollectSinkOperatorCoordinator coordinator = new CollectSinkOperatorCoordinator()) {
+            coordinator.start();
+
+            try (final TestingSocketServer socketServer =
+                    TestingSocketServer.createSocketServerAndInitializeCoordinator(coordinator)) {
+                final String version = "version";
+                final CompletableFuture<CoordinationResponse> responseFuture =
+                        coordinator.handleCoordinationRequest(
+                                createRequestForClientGeneratedResponse(version));
+                assertThat(responseFuture).isNotDone();
+
+                final CompletableFuture<List<Row>> dataFuture = new CompletableFuture<>();
+                final CompletableFuture<Void> serverSideProcess =
+                        socketServer.handleRequestAsync(dataFuture);
+                assertThat(serverSideProcess).isNotDone();
+
+                // wait for the connection to be established
+                socketServer.waitForConnectionToBeEstablished();
+                // in order to close the connection from the server's side
+                socketServer.closeAcceptingSocket();
+
+                assertEmptyResponseGeneratedFromClient(responseFuture, version);
+
+                assertThat(serverSideProcess).isNotDone();
 
-            // server closes here
-            failedResponseFuture =
+                dataFuture.complete(Collections.singletonList(Row.of(123, "abc")));
```

Comment:
why do we touch the data-future again? It's a server-side thing and should have no effect on the test results.

Reply:
fair enough. I removed these lines.